### PR TITLE
Temporarily disable the parallelism in make for the staging step

### PR DESCRIPTION
**Pull request by wallrj**
This is an attempt to fix the error:

> cp: cannot create regular file '_bin/scratch/manifests/deploy/chart/cert-manager-v1.11.0-alpha.0.tgz.prov': No such file or directory

We suspect that there may be some missing dependency in one of the make targets, or a problem with files not being synced to disk.
See https://kubernetes.slack.com/archives/CDEQJ0Q8M/p1668590557498289 for the full discussion.

This slowed the staging step from 5 min to ~12min

### CODE REVIEW

File: gcb/makestage/cloudbuild.yaml
Comment (SgtCoDFish) [inline gcb/makestage/cloudbuild.yaml:24]: suggestion: If we're gonna reduce parallelism as a solution (rather than try to fix the issue for next time) should we reduce the size of the machine we're using (`options.machineType` at the bottom) to a smaller type with fewer cores? Feels excessive having 32 cores if we only use one!

**Review by SgtCoDFish — CHANGES_REQUESTED** [review]
Got a suggestion if we take this approach!

I think I could throw together a PR which might help fix the actual issue pretty quickly if we want to take a different approach. Let me know what you think!

File: gcb/makestage/cloudbuild.yaml
Comment (wallrj) [reply gcb/makestage/cloudbuild.yaml:24]: The 32 cores will must help speed up the go builds so I'd prefer to make the minium viable change here and revert it once we've fixed the underlying issue.

**Comment by wallrj** [comment]
I'm in favour of fixing the underlying issue, but also don't want to hold up the patch releases. 
Presumably any fix to the Makefiles would also need to be backported to 1.9 and 1.10 which would delay things.

So I would prefer to merge this and revert it once we have fixed the Makefile.

**Comment by wallrj** [comment]
Or, I suppose I can continue to use the local change without comitting it to the master branch.

File: gcb/makestage/cloudbuild.yaml
Comment (SgtCoDFish) [reply gcb/makestage/cloudbuild.yaml:24]: That's reasonable, yeah!

**Review by SgtCoDFish — APPROVED** [review]
I think your suggestion to use the local change is a good one and I'd be happy with that. If we merge this it's easy for it to be forgotten about!

I'm gonna approve this anyway in case you'd like to merge this, so I don't get in the way :grin:

/lgtm
/approve

**Comment by SgtCoDFish** [comment]
I've found the reason this happens!

On [this line](https://github.com/cert-manager/cert-manager/blob/43e13bfa0dce9184681621a7e85b769276e015fb/make/manifests.mk#L51) and [this line](https://github.com/cert-manager/cert-manager/blob/43e13bfa0dce9184681621a7e85b769276e015fb/make/manifests.mk#L60) we remove a directory inside a make target, and both targets rely on (and work with) that directory, causing a race condition.

Both targets should use different directories if they're gonna rely on creating / removing directories during execution.

I'm gonna close this PR and work on a solution to the underlying issue!

**Pull request closed without merging**
